apisix/balancer.lua

```diff
@@ -134,6 +145,30 @@ local function pick_server(route, ctx)
         end
     end
 
+    local retries = up_conf.retries
+    if not retries or retries < 0 then
+        retries = #up_conf.nodes - 1
```

Comment:
Is it possible that nodes have a length of 0 ？

Reply:
This case is rejected early:
https://github.com/apache/apisix/blob/c9d3cba845673ba55626890fb08ca28f44f8a24c/apisix/upstream.lua#L268-L271